src/client/Microsoft.Identity.Client/OAuth2/TokenClient.cs

```diff
@@ -157,8 +158,11 @@ private void AddBodyParamsAndHeaders(IDictionary<string, string> additionalBodyP
                     _serviceBundle.HttpTelemetryManager.GetLastRequestHeader());
             }
 
-            //Signaling that the client can perform PKey Auth
-            _oAuth2Client.AddHeader(PKeyAuthConstants.DeviceAuthHeaderName, PKeyAuthConstants.DeviceAuthHeaderValue);
+            //Signaling that the client can perform PKey Auth on supported platforms
+            if (DeviceAuthHelper.CanOSPerformPKeyAuth())
+            {
+                _oAuth2Client.AddHeader(PKeyAuthConstants.DeviceAuthHeaderName, PKeyAuthConstants.DeviceAuthHeaderValue);
+            }
```

Comment:
@pmaytak We used to advertise that we are PKey Auth capable on every call in the token client but now we are looking to only do this on older versions of windows. (win 8 and below).

Thus I am adding these checks here to make sure we are only adding this header on supported platforms. Do you think these checks will have any impact on performance?